### ISSUE TITLE
What automatic checks are done when uploading a photo?

### ISSUE THREAD

**Issue by BeckiBorastonBucks** [alphagov/verify-local-patterns#746]
"Your photo has passed our automatic checks." What automatic checks were done? What do we need to build to include the same checks?

https://verify-local-patterns.herokuapp.com/service-patterns/concessionary-travel/example-service/webcam-photo-accepted

https://verify-local-patterns.herokuapp.com/public/images/ctscreenshot/photo-received.png

**Comment by EUzkuraityte** [comment]
@sanjaypoyzer @lizziebruce @samuelculley could you provide an overview of the checks being done?

I would presume it is to do with picture size?

**Comment by lizziebruce** [comment]
@EUzkuraityte my presumption was that software auto-detects a face?

**Comment by EUzkuraityte** [comment]
@BeckiBorastonBucks Indeed, our prototype doesn't have the ability to detect the face. If that's something that can be built for you easily, brilliant, if not, just file size will suffice.

**Comment by ghost** [comment]
@BeckiBorastonBucks We don't want to prescribe what checks need to be done, but there are normally checks that revolve around the size and format of the file, and making sure it's not a virus being uploaded to your site. If you've got any other online services that involve uploading a file, you might be able to copy the sort of thing they're doing. 

For face recognition, you have to be quite careful, as a lot of software that does this isn't up to a standard that government is comfortable with - it's quite easy to get software that doesn't consistently recognise people in various demographic groups, and that can be interpreted as indirect discrimination!

Let us know if you need any more info, we can try and find out what other councils are up to, or even teams within GDS.
